src/Standards/Generic/Sniffs/CodeAnalysis/EmptyPHPStatementSniff.php

```diff
@@ -53,12 +53,21 @@ public function process(File $phpcsFile, $stackPtr)
         case 'T_SEMICOLON':
             $prevNonEmpty = $phpcsFile->findPrevious(Tokens::$emptyTokens, ($stackPtr - 1), null, true);
 
-            if ($prevNonEmpty === false
-                || ($tokens[$prevNonEmpty]['code'] !== T_SEMICOLON
+            if ($prevNonEmpty === false) {
+                return;
+            }
+
+            if ($tokens[$prevNonEmpty]['code'] !== T_SEMICOLON
                 && $tokens[$prevNonEmpty]['code'] !== T_OPEN_TAG
-                && $tokens[$prevNonEmpty]['code'] !== T_OPEN_TAG_WITH_ECHO)
+                && $tokens[$prevNonEmpty]['code'] !== T_OPEN_TAG_WITH_ECHO
             ) {
-                return;
+                // Still detect `if (foo) {};`.
+                if ($tokens[$prevNonEmpty]['code'] !== T_CLOSE_CURLY_BRACKET
+                    || isset($tokens[$prevNonEmpty]['scope_condition']) === false
+                    || in_array($tokens[$tokens[$prevNonEmpty]['scope_condition']]['code'], [T_CLOSURE, T_ANON_CLASS]) === true
```

Comment:
Nitpick/micro-optimization: a straight comparison will be faster than a call to `in_array()`.
```php
                // Still detect `if (foo) {};`.
                if ($tokens[$prevNonEmpty]['code'] !== T_CLOSE_CURLY_BRACKET
                    || isset($tokens[$prevNonEmpty]['scope_condition']) === false
                ) {
                    return;
                }

                $scopeOwner = $tokens[$tokens[$prevNonEmpty]['scope_condition']]['code'];
                if ($scopeOwner === T_CLOSURE || $scopeOwner === T_ANON_CLASS) {
                    return;
                }
```